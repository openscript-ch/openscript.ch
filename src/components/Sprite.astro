---
import SpriteData from "../../res/sprite/sprite.json";

export type Props = {
  name: keyof typeof SpriteData;
  class?: string;
  viewBox?: string;
};

const dimensions = SpriteData[Astro.props.name];
const { class: className, viewBox, name: spriteName } = Astro.props;
---

<svg
  class={className ?? undefined}
  width={viewBox ? undefined : dimensions.width}
  height={viewBox ? undefined : dimensions.height}
  viewBox={viewBox}
>
  <use href={`/sprite.svg#${spriteName}`}></use>
</svg>
