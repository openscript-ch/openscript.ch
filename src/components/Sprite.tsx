import SpriteData from '../../static/sprite.json';

type SpriteProps = {
  name: keyof typeof SpriteData;
  className?: string;
};

export function Sprite({ name, className }: SpriteProps) {
  const dimensions = SpriteData[name];

  return (
    <svg version="2.0" className={className ?? undefined} width={dimensions.width} height={dimensions.height}>
      <use href={`/sprite.svg#${name}`} />
    </svg>
  );
}
