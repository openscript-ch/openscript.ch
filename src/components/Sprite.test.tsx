import { render } from '@testing-library/react';
import { Sprite } from './Sprite';

describe('<Sprite />', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Sprite name="arrow" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
