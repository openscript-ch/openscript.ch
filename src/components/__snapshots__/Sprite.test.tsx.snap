// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sprite /> should match snapshot 1`] = `
<DocumentFragment>
  <svg
    height="10"
    version="2.0"
    width="9.387"
  >
    <use
      href="/sprite.svg#arrow"
    />
  </svg>
</DocumentFragment>
`;
